### DISCUSSION TITLE
Docker/Cloud deployment for simulation

### DISCUSSION THREAD

**Discussion by abelsiqueira** [asreview/asreview#1397] [Ideas]
Hi @Rensvandeschoot, @J535D165, @jteijema, @jspaaks,

I will formalize the initial plans for "one-click" cloud deployment here and gather user stories. This is based on discussions that I had with each of you, in some cases separately. Here is the current idea:

- Split Docker images to serve specific purposes (base, web app, simulation, possibly others).
- The web app image can be used on SURF RC, internal servers, or self-hosted. It serves the same purpose of the older asreview image.
- The web app one-click deployment would be a pre-configuration of the docker instance on the cloud - this is what the current implementation is doing. They require cloud availability and configuration as well as knowledge of how to access the cloud-hosted web app. This means that it is not easy enough for a non-technical user to use the one-click button. It would benefit a somewhat technical user, but the most benefit would be from having a SURF RC solution. For these reasons, we discussed not continuing with the line of development.
- The CLI image is built from all official tools except for the web app.
- The CLI image is to be used by technical users, i.e., someone who knows how to access some cloud computing. For instance, `ssh`ing into an AWS machine, running the simulation, and then getting the result back into their machine. For this reason, a graphical button that does some of the jobs would be of little value. And a complete GUI for the simulation is out of scope.
- Based on the previous comments, the "one-click" deployment for simulation would be something that increases the productivity of technical users to run their simulations on the cloud and get the results back. Ideally, we want to keep the tool agnostic, but there are some benefits to knowing the cloud platform (e.g., some have CLIs to manage VMs).
- It is not clear yet whether this tool will run makita, be orthogonal to makita, be an alternative to makita, or be part of makita. 
- There is also a possibility of integrating the asreview tool with synergy directly. It is too early to discuss.
- There is also a discussion on splitting asreview into two or more parts. It would benefit the separate image building, but in principle, it is not a requirement.

This image summarizes the dependencies and docker images:

![asreview-docker](https://user-images.githubusercontent.com/1068752/231414328-77bfd218-566f-4dcf-8640-c7d9e6ad0565.jpg)

User stories for cloud tool (called asreview-cloud for now):

- "I want to run the jobs.sh file that makita gives me, but I don't want to do it on my computer. Just run on the cloud (here's the IP, login, and password), and send me an e-mail when it's done."
- "I want to verify the performance of a new extractor/classifier against all other extractor/classifiers on the synergy dataset. This would probably take days/weeks if it ran sequentially, so I want to spin up several instances of the simulator and aggregate the results."
- "I have a collection of datasets for my field, and I want to fine-tune a model (i.e., find the hyperparameters) for this dataset. I don't want the details, just give me what to use. Here's an IP, login and password." - I guess this is just a variation of the first?

**Comment by J535D165** [comment]
Thanks for the overview. I like the SYNERGY image. Can't we use our current image as the base image of `asreview-cli`? Once ASReview is split, we can adapt to these changes.

**Reply by abelsiqueira** [reply]
Our current image includes everything, web and cli.
